src/hooks/v2/transactor/RedeemTokensTx.ts

```diff
@@ -0,0 +1,48 @@
+import { NetworkContext } from 'contexts/networkContext'
+import { BigNumber } from '@ethersproject/bignumber'
+import { randomBytes } from '@ethersproject/random'
+import { useContext } from 'react'
+
+import { V2UserContext } from 'contexts/v2/userContext'
+import { V2ProjectContext } from 'contexts/v2/projectContext'
+
+import { TransactorInstance } from '../../Transactor'
+
+export function useRedeemTokensTx(): TransactorInstance<{
+  redeemAmount: BigNumber
+  minAmount: BigNumber
+  preferConverted: boolean
+}> {
+  const { transactor, contracts } = useContext(V2UserContext)
+  const { userAddress } = useContext(NetworkContext)
+  const { projectId } = useContext(V2ProjectContext)
+
+  return ({ redeemAmount, minAmount, preferConverted }, txOpts) => {
+    if (
+      !transactor ||
+      !userAddress ||
+      !projectId ||
+      !contracts?.JBETHPaymentTerminal
+    ) {
+      txOpts?.onDone?.()
+      return Promise.resolve(false)
+    }
+
+    const memo = '' //TODO: memo field on V2RedeemModal
+
+    return transactor(
+      contracts?.JBETHPaymentTerminal,
+      'redeemTokensOf',
+      [
+        userAddress,
+        projectId.toHexString(),
```

Comment:
@johnnyd-eth typically `ethers` will handle any conversions for us, we can pass it BigNUmbers and it's generally all g. Might fuck up sometimes so good to check.